### ISSUE TITLE
When a divided road truck route meets a non-truck route at an intersection, divided road connectors may need to be tagged as truck routes

### ISSUE THREAD

**Issue by mraross** [bcgov/ols-router#201]
This is very subtle so here is an example:
Using the router demo app, 
1. Turn the truck routes layer on

2. Set follow truck route to true and truck route multiplier to 90

3. Request a route from:

       56.23007,-120.82141
to

       56.21752,-120.79445

The resulting route should turn left at hwy 97. Instead, it turns right at hwy 97 goes in the wrong direction for 800m then turns around, goes back to 86th St then carries on in the correct direction toward the destination. This adds about two pointless kilometers to the trip. Making the truck route multiplier very large (e.g., 1700000000000) has no effect on the route. Reducing the multiplier below 80 fixes the problem but this value is too low to draw trucks to the nearest truck route in the general case.

The root cause of the problem is that the connector on 86th between the hwy 97 road segments is not tagged as as a truck route.

So, enhance Routable BC Maker to detect such connectors and tag them as truck routes.

**Comment by mraross** [comment]
This could be treated purely as an ITN data issue if no algorithmic way to identify and tag such segments can be found.

**Comment by gleeming** [comment]
Chris and I have discussed the option of considering these road connectors as pseudo truck routes.  If they can be identified for the router, they would be treated internally like truck routes for routing purposes but would not be considered as part of the truck route network for output directions and partitioning.

This topic came up with TransLink yesterday.  The image below shows a non-truck road (southern east/west grey one) joining into a divided road.  The intermedian connector is correctly set in ITN as a non truck route, but currently will not act like one for routing.

To extend on this concept, TransLink would like to standardize protected left turn lanes that transition from inside a divided truck route onto a non truck route as per the marked segment in the image.  However they are concerned that globally changing these to non-truck routes in the ITN will affect routing in the current release.

I have recommended that we do the opposite in the data for now, i.e. standardize these left-turn lanes to truck routes and keep track of them.  If an app change as per this ticket is made to automatically deal with these cases, then the data can be reverted.

![image](https://user-images.githubusercontent.com/21045965/82581187-0bb9dc00-9b45-11ea-9e2a-65030e5f2d74.png)
